### ISSUE TITLE
[FEATURE] Add machine learning examples with scikit-learn

### ISSUE THREAD

**Issue by hackdartstorm** [hackdartstorm/Python#270]
## 📋 Issue Description

Create beginner-friendly machine learning examples.

## 🔍 Current State

Only LLM-focused ML exists, need basic ML examples.

## ✅ Expected Behavior

Progressive ML examples:

### Example 1: Classification
- Load dataset (Iris/Wine)
- Train/test split
- Train classifier
- Evaluate accuracy
- Visualize results

### Example 2: Regression
- Load dataset (Housing)
- Feature engineering
- Train regressor
- Evaluate (MSE, R²)
- Predict new values

### Example 3: Clustering
- Load dataset
- K-Means clustering
- Visualize clusters
- Evaluate silhouette score

### Example 4: Pipeline
- Preprocessing
- Feature selection
- Model training
- Cross-validation

## 📝 Tasks

- [ ] Create ml_examples/ folder
- [ ] Add requirements.txt
- [ ] Create 4 example notebooks/scripts
- [ ] Include datasets or download code
- [ ] Add visualizations
- [ ] Create README with explanations
- [ ] Add ML best practices guide

## 💡 Example: Classification

```python
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score, classification_report

# Load data
iris = load_iris()
X, y = iris.data, iris.target

# Split data
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

# Train model
clf = RandomForestClassifier(n_estimators=100)
clf.fit(X_train, y_train)

# Predict
y_pred = clf.predict(X_test)

# Evaluate
print(f"Accuracy: {accuracy_score(y_test, y_pred):.2f}")
print(classification_report(y_test, y_pred))
```

## 📝 Difficulty

**Intermediate** - ML basics required

## 🔗 Affected Files

- New: projects/ml_examples/*
